src/js/services/request.js

```diff
@@ -27,10 +27,9 @@ angular.module('app')
       // utils
       function setOptions(options) {
         options.headers = options.headers || {};
-        options.timeout = options.timeout || 5000;
+        options.timeout = options.timeout || 20000;
```

Comment:
Maybe it'd be better to have this configurable in a config file rather than hard coded in code, but that's something which could be added in another PR.